### ISSUE TITLE
Steps for React Native Compat

### ISSUE THREAD

**Issue by Un3qual** [gregberge/xstyled#123]
## 🚀 Feature Proposal
Most of the issues with React Native compatibility come from the color modes update because it uses DOM manipulation and CSS custom properties which don't work on React Native

## Motivation
React Native compatibility would make this library even better and have more use cases

## Example/How
Most of the changes need to made in `core/colorModes.js`. All of the differences would require detecting React Native, either through different imports like how `styled-components` does it with `styled-components/native` or with detecting `window.navigator`. Please note that I have not tested any of the example code, it is just to start a discussion on how React Native compatibility would be implemented. 

### Storage
The current storage implementation could be modified to use async-storage
https://github.com/smooth-code/xstyled/blob/fc70500b994634a0aa3d79e4d5fdaa1cc531c711/packages/core/src/colorModes.js#L11-L18
When React Native is detected, storage could use 
```javascript
import AsyncStorage from '@react-native-community/async-storage'

const storage = { 
   get: async () => {
      try {
         AsyncStorage.getItem(`@${STORAGE_KEY}`)
      } catch (e) {
         console.error('error')
      }
   }, 
   set: async (value) => {
      try {
         AsyncStorage.setItem(`@${STORAGE_KEY}`, value)
      } catch (e) {
         console.error('error')
      }
   }, 
   clear: () => {
      try {
         AsyncStorage.removeItem(`@${STORAGE_KEY}`)
      } catch (e) {
         console.error('error')
      }
   }, 
 } 
```

### Detecting System Mode
For detecting the system color mode, `react-native-appearance` could be used. 

## Pitch
Xstyled improves on styled-system by making it easier to use by allowing the system to be used in css`` blocks. This would allow those features to be used in React Native, further improving ease of use.

**Comment by diegotsi** [comment]
Hi @Un3qual! The version `1.11.0` works pretty well on react-native but the others do not.

**Comment by gregberge** [comment]
Re-open a new one for v2 please.
